spec/multisite/s3_store_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+require 'file_store/s3_store'
+
+RSpec.describe 'Multisite s3 uploads', type: :multisite do
+  let(:conn) { RailsMultisite::ConnectionManagement }
+  let(:store) { FileStore::S3Store.new }
+  let(:s3_helper) { store.instance_variable_get(:@s3_helper) }
+  let(:uploaded_file) { file_from_fixtures("logo.png") }
+
+  before(:each) do
+    SiteSetting.s3_upload_bucket = "s3-upload-bucket"
+    SiteSetting.s3_access_key_id = "s3-access-key-id"
+    SiteSetting.s3_secret_access_key = "s3-secret-access-key"
+    SiteSetting.enable_s3_uploads = true
+  end
+
+  shared_context 's3 helpers' do
+    let(:store) { FileStore::S3Store.new }
+    let(:client) { Aws::S3::Client.new(stub_responses: true) }
+    let(:resource) { Aws::S3::Resource.new(client: client) }
+    let(:s3_bucket) { resource.bucket("s3-upload-bucket") }
+    let(:s3_helper) { store.instance_variable_get(:@s3_helper) }
+  end
+
+  context 'uploading to s3' do
+    include_context "s3 helpers"
+
+    describe "#store_upload" do
+      it "returns the correct url for default multisite db" do
+        s3_helper.expects(:s3_bucket).returns(s3_bucket).at_least_once
```

Comment:
I know I did this in the past as well but @gschlager had showed me a new way of testing these. Can you have a look at https://github.com/discourse/discourse/blob/c27502793901767d584f28e254caa7de000363c2/spec/lib/backup_restore/s3_backup_store_spec.rb and see if we can adopt that pattern instead of using the mocks here?